examples/js/controls/DeviceOrientationControls.js

```diff
@@ -61,8 +61,27 @@ THREE.DeviceOrientationControls = function ( object ) {
 
 		onScreenOrientationChangeEvent(); // run once on load
 
-		window.addEventListener( 'orientationchange', onScreenOrientationChangeEvent, false );
-		window.addEventListener( 'deviceorientation', onDeviceOrientationChangeEvent, false );
+		if ( typeof window.DeviceMotionEvent.requestPermission === 'function' ) {
```

Comment:
In order to make the code more robust, write this statement like so:
```js
if ( window.DeviceMotionEvent !=== undefined && typeof window.DeviceMotionEvent.requestPermission === 'function' ) {
```
When using the example without `https`, your code introduces a runtime error otherwise.